amity/amity.py

```diff
@@ -0,0 +1,353 @@
+            self.waiting_list['office'].remove(person_name)
+            cprint("{} removed from waiting list".format(
+                   person_name), "cyan")
+
+    def remove_person_from_previous_livingspace(self, person_name):
+        """ Removes a fellow from the living spaces they were previously
+            located in or from the waiting list if they did not have a
+            livingspace """
+        for room in self.all_rooms['livingspace']:
+            for person in room.room_occupants:
+                if person == person_name:
+                    room.room_occupants.remove(person_name)
+                    cprint("{} removed from previous livingspace"
+                           .format(person_name), "cyan")
+                    return
+        if person_name in self.waiting_list['livingspace']:
+            self.waiting_list['livingspace'].remove(person_name)
+            cprint("{} removed from waiting list".format(
+                   person_name), "cyan")
+
+    def reallocate_staff(self, person_name, new_room_name):
+        """ Reallocate a staff to a different office and rejects allocation to
+            a living space """
+        try:
+            for room in self.all_rooms['office'] + self.all_rooms[
+                                                               'livingspace']:
+                if room.room_name == new_room_name \
+                        and room.room_type == "office":
+                    for person in room.room_occupants:
+                        # checks if person already exists in the new office
+                        if person.person_name == person_name:
+                            cprint("{} already exists in the room {}".format(
+                                   person_name, new_room_name), "red")
+                            return
+                    # reallocate a person and call function
+                    # remove_person_from_previous_office
+                    # to remove the staff from the previous
+                    # office or waiting list
+                    if len(room.room_occupants) < 6:
+                        self.remove_person_from_previous_office(person_name)
+                        room.room_occupants.append(person_name)
+                        cprint("{} has been reallocated to office {}" .format(
+                               person_name, new_room_name), "cyan")
+                        return
+                    else:
+                        cprint("Office {} is full, choose another room".format(
+                               new_room_name), "red")
+
+                elif room.room_name == new_room_name \
+                        and room.room_type == "livingspace":
+                    cprint("Cannot reallocate a staff member to a living space", "red")
+                    return
+        except ValueError:
+            cprint("Wrong inputs", "red")
+
+    def reallocate_fellow(self, person_name, new_room_name):
```

Comment:
These two functions do the same thing. Combine them into one `reallocate_person` and pass the `person_type` as an argument.